src/extension/commands_helpers.c

```diff
@@ -165,6 +171,32 @@ static inline dd_result _omsg_send(dd_conn *nonnull conn, dd_omsg *nonnull omsg)
     return dd_conn_sendv(conn, &omsg->iovecs);
 }
 
+#ifdef SO_PASSCRED
+static inline dd_result _omsg_send_cred(
+    dd_conn *nonnull conn, dd_omsg *nonnull omsg)
+{
+    // set SO_PASSCRED before sending the message. This is to try to
+    // ensure that the helper does not send a response ahead of our having
+    // had the chance to set SO_PASSCRED before calling recvmsg(), resulting in
+    // the credentials received having the overflowuid
+    int res = setsockopt(
+        conn->socket, SOL_SOCKET, SO_PASSCRED, &(int){1}, sizeof(int));
```

Comment:
The only comment (nitpick) I have is that when `check_cred` is set, there will be two calls to setsockopt to enable `SO_PASSCRED`.

Reply:
That was an oversight caused by them being in different files. I've added a new commit to fix it and move everything to network.c